ui.h

```diff
@@ -99,7 +99,7 @@ struct Ui {
 	void (*suspend)(Ui*);
 	void (*resume)(Ui*);
 	bool (*getkey)(Ui*, TermKeyKey*);
-	void (*terminal_save)(Ui*);
+	void (*terminal_save)(Ui*, bool fscr);
```

Comment:
Since you have changed the function signature here you need to change it in all the backends. Right now this breaks compilation for the vt100 backend.

Reply:
True, thanks for spotting this!
I updated the patch, accordingly.

I used vis_pipe() for two reasons:
1. I wanted to avoid a separate function in the API
2. I think, you should also be able to pipe through interactive programs, e.g. a fuzzy finder. While there are commands in sam.c for different piping scenarios (cmd_filter(), cmd_pipein(), cmd_pipeout(), cmd_launch()) which boil down to calling vis_pipe(), in the Lua API this can be handled by one function, so no specialization is needed. Current plugins tend to use the Lua standard io.popen() for that purpose (e.g. https://github.com/peaceant/vis-fzf-mru.git). However, this relies on the external command to not change the terminal settings, so it's safer to use vis:pipe(), instead.